### PR TITLE
Nearcache Refactor

### PR DESCRIPTION
**Pull request by yuce**
Same content with `nearcache-impl` branch with different layout.

### CODE REVIEW

File: internal/nearcache/record.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import (
+	"sync/atomic"
+
+	"github.com/hazelcast/hazelcast-go-client/internal"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+type Record struct {
+	CreationTime         int64
```
Comment (utku-caglayan) [inline internal/nearcache/record.go:27]: commenting to point out the field align warning

File: internal/nearcache/record_store.go [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client/internal/serialization"
+	"github.com/hazelcast/hazelcast-go-client/nearcache"
+)
+
+type RecordStore struct {
+	stats            Stats
```
Comment (utku-caglayan) [inline internal/nearcache/record_store.go:30]: commenting to point out the field align warning

File: internal/nearcache/stats.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * limitations under the License.
+ */
+
+package nearcache
+
+import "time"
+
+// Stats contains statistics for a Near Cache instance.
+type Stats struct {
+	InvalidationRequests int64
```
Comment (utku-caglayan) [inline internal/nearcache/stats.go:23]: commenting to point out the field align warning

**Review by utku-caglayan — APPROVED** [review]

**Comment by yuce** [comment]
Redirecting to the feature branch,

**Pull request closed without merging**